src/v/cluster/config_manager.cc

```diff
@@ -821,6 +811,14 @@ config_manager::apply_delta(cluster_config_delta_cmd&& cmd_in) {
           _seen_version);
         co_return errc::success;
     }
+    _seen_version = delta_version;
+    // version_shard is chosen to match controller_stm_shard, so
+    // our raft0 stm apply operations do not need a core jump to
+    // update the frontend version state.
+    vassert(
+      ss::this_shard_id() == config_frontend::version_shard,
+      "Must be called on frontend version_shard");
+    _frontend.local().set_next_version(_seen_version + config_version{1});
```

Comment:
I understood the intent of this change but I have a dumb question...

>  leading to a situation in which the configuration was overridden by the previous update.

What exactly is reordering the updates? I skimmed thru mux_state_machine code and it should always `apply_update` the deltas in order after each returns, no?

Reply:
exactly, this is correct, the problem is in the configuration_frontend which is managing and validating version